lib/ruby/lib/alces/stack/templater.rb

```diff
@@ -45,99 +46,108 @@ def append(template_file, append_file, template_parameters={})
           end
         end
 
-        def replace_hash(template, count, template_parameters={})
-          raise "Templater loop count reached. Check parameters for infinite loops" if count > 100
-          tags = template.scan(/<%=[ \w\*\+\-\/\%\(\)\=\!]*%>/)
-          return template if tags.length == 0
-          error_tag = false
-          error_message  = "Could not find value(s) for:"
-          tags.each do |t|
-            t.scan(/_*[[:alpha:]][[:alnum:]_]*/) do |word|
-              if !template_parameters.has_key?(word.to_sym)
-                error_tag = true
-                error_message << "\n  " << t
-                break
-              end
-            end
+        def replace_erb(template, template_parameters={})
+          return ERB.new(template).result(OpenStruct.new(template_parameters).instance_eval {binding})
+        rescue StandardError => e
+          $stderr.puts "Could not parse ERB"
+          $stderr.puts template.to_s
+          $stderr.puts template_parameters.to_s
+          raise e
+        end
+      end
+
+      class Combiner < Handler
+        def initialize(json, hash={})
+          @combined_hash = {
```

Comment:
This is always going to be the same -- perhaps it could be pulled out into a constant:

```
class Combiner < Handler
  DEFAULT_PARAMS = { hostip: `hostname -i`.chomp, index: 0 }
  def initialize(json, hash = {})
    @combined_hash = DEFAULT_PARAMS.merge(hash)
    [...]
```